xbmc/filesystem/FileCache.cpp

```diff
@@ -234,8 +234,36 @@ void CFileCache::Process()
     return;
   }
 
+  // How much should we request? Well if we have a front buffer size, use that.
+  // Note however that if more than half of the forward buffer is empty,
+  // that would mean that we're falling behind and if we try to read more
+  // than what is in the buffer now in that situation, we will run out of bytes
+  // before the next read call completes. Hence it never makes sense to read
+  // more than half of the forward cache size at a time as even if that were
+  // to improve throughput, we have too little buffer space to keep up.
+  int64_t requestSize;
+  if (m_forwardCacheSize)
+  {
+    requestSize = m_forwardCacheSize / 2;
+  }
+  else
+  {
+    // We have "unlimited" buffer capabilities, but we still must bound the
+    // request size to something.
+    // Guiding light: If sufficient bandwidth is available, things
+    // should work even if the latency is large, for distant servers.
+    // For example: Bluray Ultra HD can reach a bitrate of about 144Mbit/s.
+    // So that's roughly 20MiB/s. In a high throughput/high latency network,
+    // one read roundtrip might might take 500ms.
+    // At steady state, we must read at least as much as the player
+    // consumes over a given time period, so during 500ms we must
+    // read 10MiB. This is obviously a simplified view of the network,
+    // but for now it's better than nothing.
+    requestSize = 10 * 1024 * 1024;
```

Comment:
Please add this as a constexpr int in an anonymous namespace at the start of the file.

```
namespace
{

constexpr int DEFAULT_REQUEST_SIZE = 10 * 1024 * 1024;

}
```